### PR TITLE
tooltips on copilot

### PR DESCRIPTION
**Pull request by SanderGi**
Task: https://app.asana.com/0/1203067047205953/1206050253347928

Server PR: https://github.com/GooeyAI/gooey-server/pull/348

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208027661017554

### CODE REVIEW

**Comment by devxpy** [comment]
Hmm, can't we just do the tooltip on the `RenderedMarkdown` widget, as opposed to defining the same thing on every input component?

In server code too then, we can pass the tooltip with `st.write`

[DRY!](https://wiki.c2.com/?DontRepeatYourself)

**Comment by SanderGi** [comment]
> Hmm, can't we just do the tooltip on the `RenderedMarkdown` widget, as opposed to defining the same thing on every input component?
> 
> In server code too then, we can pass the tooltip with `st.write`
> 
> [DRY!](https://wiki.c2.com/?DontRepeatYourself)

We can, but it will require an extra layer in terms of a wrapper element to get the styling right. Shouldn't be a big performance hit since modern browsers are fast, but it is less readable. It is also a bit weird since we use the markdown component for the tooltip. Should I still move it to the markdown component?

We can't keep it in server code local to `st.write` because the inputs don't use st.write for their label.

**Comment by SanderGi** [comment]
Position absolute does not work with display: contents, so I had to use the flexbox method

**Comment by anish-work** [comment]
@devxpy PR is Updated with new changes

File: app/renderedMarkdown.tsx [hunk 1]
```diff
@@ -1,17 +1,51 @@
...
+          : " gui_tooltip--right")
+      }
+    >
+      <i
+        role="button"
+        className="fa-regular fa-circle-info"
+        style={{ height: "16px", ...props.iconStyle }}
+      ></i>
+      <span className="tooltip_text">
+        <RenderedMarkdown body={props.text} className="" />
```
Comment (devxpy) [inline app/renderedMarkdown.tsx:24]: This dependency loop is a bit confusing, can you refactor to have 3 widgets  perhaps, RenderedMarkdownRaw (w/o tooltip), RenderedMarkdown and GooeyTooltip ?

**Comment by devxpy** [comment]
we need a popover to implement the new workspace switcher and hence considering to ditch this and implement both popover and tooltip using https://github.com/floating-ui/floating-ui

**Pull request closed without merging**